File: Ontology/SubClassOf.cs
namespace Ontology
{
    public class SubClassOf:
        Axiom,
        ISubClassOf
    {
        private readonly IClassExpression _subClassExpression;
        private readonly IClassExpression _superClassExpression;

        public SubClassOf(
            IOntology        ontology,
            IClassExpression subClassExpression,
            IClassExpression superClassExpression
            ): base(ontology)
        {
            _subClassExpression   = subClassExpression;
            _superClassExpression = superClassExpression;
        }

        IClassExpression ISubClassOf.SubClassExpression => _subClassExpression;

        IClassExpression ISubClassOf.SuperClassExpression => _superClassExpression;
    }
}

File: Web/ClientApp/src/app/Ontology/SubClassOf.ts
import { Axiom } from "./Axiom";
import { IAxiomSelector } from "./IAxiomSelector";
import { IAxiomVisitor } from "./IAxiomVisitor";
import { IClassExpression } from "./IClassExpression";
import { IOntology } from "./IOntology";
import { ISubClassOf } from "./ISubClassOf";

export class SubClassOf
    extends Axiom
    implements ISubClassOf
{
    constructor(
        ontology                   : IOntology,
        public SubClassExpression  : IClassExpression,
        public SuperClassExpression: IClassExpression
        )
    {
        super(ontology);
    }

    Accept(
        visitor: IAxiomVisitor
        )
    {
        visitor.SubClassOf(this);
    }

    Select<TResult>(
        selector: IAxiomSelector<TResult>
        ): TResult
    {
        return selector.SubClassOf(this);
    }
}
